tests/tests_pytorch/trainer/test_trainer.py

```diff
@@ -599,7 +599,10 @@ def training_step(self, batch, batch_idx):
             output["loss"] = output["loss"] * 0.0  # force minimal loss to trigger early stopping
             self.log("loss", output["loss"])
             self.training_step_invoked += 1
-            assert not self.trainer.should_stop
+            if self.current_epoch < 2:
+                assert not self.trainer.should_stop
+            else:
+                assert self.trainer.should_stop
```

Comment:
Why is this ever True?

Reply:
because min_epochs is 5, `should_stop` will remain `True`for some time until `min_epochs` are completed.

patience is 0 here. After 1st epoch, it will store the loss. In the 2nd epoch, it will recheck it and trigger `should_stop=True`. And from 3rd to `min_epochs` it will remain `True`and will terminate after the 5th epoch.